app/src/main/java/cloudcity/Iperf3Monitor.java

```diff
@@ -0,0 +1,365 @@
+                                    }
+                                    break;
+                            }
+                        }
+
+                        public void propertyChange(PropertyChangeEvent evt) {
+                            Log.d(TAG, "--> onPropertyChange()\tchanged event: " + evt);
+
+                            switch (evt.getPropertyName()) {
+                                case "interval": {
+                                    Interval interval = (Interval) evt.getNewValue();
+                                    parseSum(interval.getSum(), defaultThroughput);
+                                    if (interval.getSumBidirReverse() != null) {
+                                        parseSum(interval.getSumBidirReverse(), defaultReverseThroughput);
+                                    }
+
+                                    Log.v(TAG, "INTERVAL\tdefaultThroughput size: " + defaultThroughput.getMeanList().size() + ", defaultReverseThroughput size: " + defaultReverseThroughput.getMeanList().size());
+                                }
+                                break;
+
+                                case "start": {
+                                    Log.v(TAG, "START\tdefaultThroughput: " + defaultThroughput + ", defaultReverseThroughput: " + defaultReverseThroughput);
+                                }
+                                break;
+
+                                case "end": {
+                                    Log.v(TAG, "END\tdefaultThroughput size: " + defaultThroughput.getMeanList().size() + ", defaultReverseThroughput size: " + defaultReverseThroughput.getMeanList().size());
+                                    Log.v(TAG, "END\tend value is: " + evt.getNewValue() + "\t\tequals END_MARKER ? " + evt.getNewValue().equals(Iperf3Parser.END_MARKER));
+
+                                    // Throughput values need to be normalized by dividing by 1e6 so => realValue = value/1e+6
+                                    double DLmin = defaultReverseThroughput.calcMin() / 1e+6;
+                                    double DLmedian = defaultReverseThroughput.calcMedian() / 1e+6;
+                                    double DLmax = defaultReverseThroughput.calcMax() / 1e+6;
+                                    double DLmean = defaultReverseThroughput.calcMean() / 1e+6;
+                                    double DLlast = getLast(defaultReverseThroughput.getMeanList()) / 1e+6;
+
+                                    double ULmin = defaultThroughput.calcMin() / 1e+6;
+                                    double ULmedian = defaultThroughput.calcMedian() / 1e+6;
+                                    double ULmax = defaultThroughput.calcMax() / 1e+6;
+                                    double ULmean = defaultThroughput.calcMean() / 1e+6;
+                                    double ULlast = getLast(defaultThroughput.getMeanList()) / 1e+6;
```

Comment:
consider extracting all this dividing into a 
```
public double normalize(double value) {
    return value / 1e+6;
}
```
method... dunno why I didn't want to do it immediatelly, but i remember not wanting to do it for some reason. can't remember which, so I was wrong.

Reply:
done, resolving.